pyquil/quil.py

```diff
@@ -786,12 +786,14 @@ def address_qubits(program, qubit_mapping=None):
 
     result = []
     for instr in program:
-        # Remap qubits on Gate and Measurement instructions
+        # Remap qubits on Gate, Measurement, and ResetQubit instructions
         if isinstance(instr, Gate):
             remapped_qubits = [qubit_mapping[q] for q in instr.qubits]
             result.append(Gate(instr.name, instr.params, remapped_qubits))
         elif isinstance(instr, Measurement):
             result.append(Measurement(qubit_mapping[instr.qubit], instr.classical_reg))
+        elif isinstance(instr, ResetQubit):
+            result.append(ResetQubit(qubit_mapping[instr.qubit]))
```

Comment:
What happens here if there are no qubits given to the `RESET`?

Reply:
Hopefully nothing, because if no qubits are given to RESET then it will create a Reset instance, not a ResetQubit instance and this code checks to see if the gate is an instance of ResetQubit. Since Reset and ResetQubit are separate types, my thinking was this would only catch RESET gates that had a specific qubit tied to them.

Reply:
Solid reasoning, IMO.